Python/bltinmodule.c

```diff
@@ -1976,9 +1976,15 @@ builtin_input_impl(PyObject *module, PyObject *prompt)
             promptstr = PyBytes_AS_STRING(po);
         }
         else {
-            po = NULL;
+            po = PyBytes_FromString("");
             promptstr = "";
         }
+        if (PyBytes_GET_SIZE(po) != (Py_ssize_t)strlen(promptstr)) {
```

Comment:
Just move this check above, in the branch `if (prompt != NULL) {`. And it would look clearer if swap operands (left -- strlen, right -- PyBytes_GET_SIZE).